server/game/core/ability/PlayerOrCardAbility.js

```diff
@@ -104,9 +104,12 @@ class PlayerOrCardAbility {
 
     buildTargetResolver(name, properties) {
         switch (properties.mode) {
-            case TargetMode.Select: return new SelectTargetResolver(name, properties, this);
-            case TargetMode.Player: return new PlayerTargetResolver(name, properties, this);
-            default: return new CardTargetResolver(name, properties, this);
+            case TargetMode.Select:
+                return new SelectTargetResolver(name, properties, this);
+            case TargetMode.Player:
+                return new PlayerTargetResolver(name, properties, this);
+            default:
```

Comment:
I tried this, and it looks like target selectors default to TargetMode.Single, but assignment of that default happens in the constructor for CardTargetResolver, which calls this.GetSelector, which calls CardSelector.for, which calls CardSelector.getDefaultedProperties. So in this function mode can in fact be unassigned, and there's a ton of cards that use that defaulting behavior.

Reply:
Ahh okay. Yeah looking at it now, TargetMode is weird because is implicitly mixes card targeting options with player / select targeting options. I guess it's fine to have this is as the default then, might be good to leave a comment on it though